packages/core/src/logger/apiLogger.ts

```diff
@@ -0,0 +1,260 @@
+    }
+  }
+
+  private applyLogResponseBody(
+    level: Level,
+    response: HttpResponse,
+    logResponse?: HttpMessageLoggingOptions
+  ) {
+    if (logResponse?.logBody) {
+      this._logger.log(level, 'Response Body: ${body}', {
+        body: response.body,
+      });
+    }
+  }
+
+  private getContentType(headers?: Record<string, string>): string {
+    return headers ? getHeader(headers, CONTENT_TYPE_HEADER) ?? '' : '';
+  }
+
+  private getContentLength(headers?: Record<string, string>): string {
+    return headers ? getHeader(headers, CONTENT_LENGTH_HEADER) ?? '' : '';
+  }
+
+  private removeQueryParams(url: string): string {
+    const queryStringIndex: number = url.indexOf('?');
+    return queryStringIndex !== -1 ? url.substring(0, queryStringIndex) : url;
+  }
+
+  private extractHeadersToLog(
+    headers?: Record<string, string>,
+    headersToInclude?: string[],
+    headersToExclude?: string[]
+  ): Record<string, string> {
+    const filteredHeaders: Record<string, string> = {};
+    if (!headers) {
+      return {};
+    }
+
+    if (headersToInclude && headersToInclude.length > 0) {
+      return this.includeHeadersToLog(
+        headers,
+        filteredHeaders,
+        headersToInclude
+      );
+    }
+
+    if (headersToExclude && headersToExclude.length > 0) {
+      return this.excludeHeadersToLog(
+        headers,
+        filteredHeaders,
+        headersToExclude
+      );
+    }
+
+    return this.removeSenstiveHeaders(headers);
+  }
+
+  private includeHeadersToLog(
+    headers: Record<string, string>,
+    filteredHeaders: Record<string, string>,
+    headersToInclude?: string[]
+  ): Record<string, string> {
+    // Filter headers based on the keys specified in headersToInclude
+    headersToInclude?.forEach((name) => {
+      const val = getHeader(headers, name);
+      if (val !== null) {
+        filteredHeaders[name] = val;
+      }
+    });
+    return this.removeSenstiveHeaders(filteredHeaders);
+  }
+
+  private excludeHeadersToLog(
+    headers: Record<string, string>,
+    filteredHeaders: Record<string, string>,
+    headersToExclude?: string[]
+  ): Record<string, string> {
+    // Filter headers based on the keys specified in headersToExclude
+    for (const key of Object.keys(headers)) {
+      if (
+        !headersToExclude?.some(
+          (excludedName) => excludedName.toLowerCase() === key.toLowerCase()
+        )
+      ) {
+        filteredHeaders[key] = headers[key];
+      }
+    }
+    return this.removeSenstiveHeaders(filteredHeaders);
+  }
+
+  private removeSenstiveHeaders(
+    headers: Record<string, string>
+  ): Record<string, string> {
+    if (this._loggingOptions.maskSensitiveHeaders) {
+      const senstiveHeaders = [
+        'Authorization',
+        'WWW-Authenticate',
+        'Proxy-Authorization',
+        'Set-Cookie',
+      ];
+      for (const key of Object.keys(headers)) {
+        if (getHeader(headers, key) !== null && senstiveHeaders.includes(key)) {
```

Comment:
Header names must be compared in a case-insensitive way.

Reply:
ok